tests/components/modbus/test_modbus.py

```diff
@@ -0,0 +1,66 @@
+"""The tests for the Modbus sensor component."""
+import pytest
+
+from homeassistant.components.modbus.const import (
+    CONF_BAUDRATE,
+    CONF_BYTESIZE,
+    CONF_PARITY,
+    CONF_STOPBITS,
+    MODBUS_DOMAIN as DOMAIN,
+)
+from homeassistant.const import (
+    CONF_DELAY,
+    CONF_HOST,
+    CONF_METHOD,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_TIMEOUT,
+    CONF_TYPE,
+)
+
+from .conftest import base_config_test
+
+
+@pytest.mark.parametrize("do_discovery", [False, True])
+@pytest.mark.parametrize("do_options", [False, True])
+@pytest.mark.parametrize("do_type", ["tcp", "serial"])
+async def test_config_modbus(hass, do_discovery, do_options, do_type):
+    """Run test for modbus."""
+    if do_type == "tcp":
```

Comment:
One of the points with parametrization is that it allows us to avoid logic in the tests. Can we make the parameters be the config instead so we can remove the if/else logic here?

Reply:
I agree, but it was easier just to type do_type :-) Changed!